src/test/java/reactor/pool/CommonPoolTest.java

```diff
@@ -2365,4 +2366,82 @@ void raceShutdownAndPreAcquiredReleaseWithEviction(Function<PoolBuilder<AtomicIn
 		assertThat(ai).as("creates and destroys stabilizes to 0").hasValue(0);
 	}
 
+	@ParameterizedTest
+	@MethodSource("allPools")
+	void maxPendingZero(Function<PoolBuilder<Integer, ?>, AbstractPool<Integer>> configAdjuster) {
+		AtomicInteger source = new AtomicInteger();
+		PoolBuilder<Integer, PoolConfig<Integer>> configBuilder = PoolBuilder
+				.from(Mono.fromSupplier(source::incrementAndGet))
+				.sizeBetween(1, 2)
+				.maxPendingAcquire(0);
+
+		AbstractPool<Integer> pool = configAdjuster.apply(configBuilder);
+
+		assertThat(pool.warmup().block(Duration.ofSeconds(1))).as("warmup").isOne();
+
+		//there is one idle resource
+		assertThat(pool.acquire().block(Duration.ofSeconds(1)))
+				.as("acquire on idle")
+				.isNotNull()
+				.hasFieldOrPropertyWithValue("poolable", 1);
```

Comment:
this makes the tests harder to support because refactoring the method/field may not get reflected in tests. Consider using `.returns(1, it -> it.getPoolable())`

Reply:
didn't know about `returns`, neat!

Reply:
the only downside is that it is "nameless" as compared to `hasFieldOrPropertyWithValue`, but the type safety and a method reference is IMO still better :)